Build fails on musl host

When compiling `perf-event-open-sys` on `x86_64-unknown-linux-musl` build fails due a difference in the function signature of `ioctl`

```
$ rustc --version
rustc 1.45.1 (c367798cf 2020-07-26)
```

```
$ cargo build
   Compiling perf-event-open-sys v0.3.1 (/home/adrianisuru/perf-event-open-sys)
error[E0308]: mismatched types
   --> src/lib.rs:110:25
    |
110 |         libc::ioctl(fd, ioctl as c_ulong, arg)
    |                         ^^^^^^^^^^^^^^^^ expected `i32`, found `u64`
    |
help: you can convert an `u64` to `i32` and panic if the converted value wouldn't fit
    |
110 |         libc::ioctl(fd, (ioctl as c_ulong).try_into().unwrap(), arg)
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: could not compile `perf-event-open-sys`.
```
On `glibc` systems the `ioctl` function signature is 
```c
extern int ioctl (int __fd, unsigned long int __request, ...) __THROW;
```
but on `musl` systems the signature is
```c
int ioctl (int, int, ...);
```
so compiling [`src/lib.rs`](https://github.com/jimblandy/perf-event-open-sys/blob/master/src/lib.rs) fails.

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -107,6 +107,10 @@ pub mod ioctls {
         ioctl: bindings::perf_event_ioctls,
         arg: A,
     ) -> c_int {
+        #[cfg(target_env = "musl")]
+        return libc::ioctl(fd, ioctl as c_int, arg);
+
+        #[cfg(not(target_env = "musl"))]
         libc::ioctl(fd, ioctl as c_ulong, arg)
     }
 }

